### PR TITLE
Template validation improvements

### PR DESCRIPTION
**Pull request by chadwhitacre**
- close tickets that are unchanged templates (#26)
- use the new `gh issue comment` cli (#25)
- serialize to work around race condition in tests (#30)

### CODE REVIEW

**Comment by chadwhitacre** [comment]
CI: https://github.com/getsentry/.github-chadwhitacre/actions/runs/528021859

File: .github/workflows/validate-new-issue-test.yml [hunk 1]
```diff
@@ -113,4 +116,52 @@ jobs:
...
+    - id: setup
+      name: setup
+      env:
+        GITHUB_TOKEN: ${{ secrets.PAT_NONMEMBER }}
+      run: |
+        issue_number=$(
+          basename $(
+            gh issue create --title "Greetings, program!" \
+              --body "
+        ### Environment
```
Comment (BYK) [inline .github/workflows/validate-new-issue-test.yml:135]: Can we fetch a template and use that verbatim instead of putting its contents here? Otherwise we'll need to remember updating this test when we change the template.

File: .github/workflows/validate-new-issue.yml [hunk 2]
```diff
@@ -27,15 +27,23 @@ jobs:
...
+          # Strip front matter. https://stackoverflow.com/a/29292490/14946704
+          sed -i'' '1{/^---$/!q;};1,/^---$/d' "$template"
           extract-headings "$template" > headings-in-template
           echo -n "$(basename $template)? "
           if [ ! -s headings-in-template ]; then
             echo "No headers in template. 🤷"
           elif [ -z "$(comm -23 headings-in-template headings-in-issue)" ]; then
             echo "Match! 👍 💃"
-            exit 0
+            if diff -Bw "$template" issue > /dev/null; then
```
Comment (BYK) [inline .github/workflows/validate-new-issue.yml:41]: Guessing `-Bw` is to detect whitespace-only changes?

File: .github/workflows/validate-new-issue.yml [hunk 3]
```diff
@@ -48,8 +56,5 @@ jobs:
 
         # Might get `gh issue comment` some day - https://github.com/cli/cli/issues/517
         echo -n "Commented: "
-        gh api "https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.issue.number }}/comments" \
-          --method POST \
-          --input comment \
-          | jq .html_url
+        gh issue comment ${{ github.event.issue.number }} --body "$(cat comment)"
```
Comment (BYK) [inline .github/workflows/validate-new-issue.yml:59]: nit: Would have been better to separate this into its own PR.

File: .github/workflows/validate-new-issue.yml (hunk 2 above)
Comment (chadwhitacre) [reply .github/workflows/validate-new-issue.yml:41]: Yup. I'll add a comment.

File: .github/workflows/validate-new-issue.yml (hunk 3 above)
Comment (chadwhitacre) [reply .github/workflows/validate-new-issue.yml:59]: Done: https://github.com/getsentry/.github/pull/34.

File: .github/workflows/validate-new-issue-test.yml (hunk 1 above)
Comment (chadwhitacre) [reply .github/workflows/validate-new-issue-test.yml:135]: Will do.

**Comment by chadwhitacre** [comment]
Merged accidentally (tightened branch protection to prevent in future). `main` reset and kicking out new PRs ...
